### PR TITLE
Feat/add limited trust test

### PR DESCRIPTION
**Pull request by mihmatache**
Updated limited trust based on solo comments.

1. Create utils package for shared logic
1. Added end 2 end test for limited trust
1. Moved certificate creation logic from the `Finished` state of the agent to the mtls_translator and `Requested` state of the cert agent.

### CODE REVIEW

File: test/e2e/env.go [hunk 1]
```diff
@@ -29,6 +29,7 @@ import (
 )
 
 const (
+	// Cluster contextx
```
Comment (john-a-joyce) [inline test/e2e/env.go:32]: Minor spelling contextx

File: test/e2e/env.go [hunk 2]
```diff
@@ -200,7 +201,7 @@ func GetEnv() Env {
 }
 
 func ClearEnv(ctx context.Context) error {
-	if useExisting := os.Getenv("USE_EXISTING"); useExisting == "1" {
+	if useExisting := os.Getenv(UseExisting); useExisting == "1" {
```
Comment (john-a-joyce) [inline test/e2e/env.go:204]: I found it odd you choose to change the name of the OS env but I don't have enough info to know if old was used elsewhere and should have been maintained.  I assume you know what you are doing.

File: test/e2e/istio/istio_e2e_suite_test.go [hunk 3]
```diff
@@ -57,14 +57,31 @@ var _ = BeforeSuite(func() {
 	federateClusters(dynamicClient)
 })
 
+// Before a test check whether limited trust is set
+var _ = BeforeEach(func() {
+	if isLimitedTrust() {
+		Skip("Limited trust does not currently support the current test case")
```
Comment (john-a-joyce) [inline test/e2e/istio/istio_e2e_suite_test.go:63]: I am not that familiar with the usage  of BeforeEach and Skip trouble seeing how checking the value of an ENV is enough to determine is a test cases supports limited trust.   Perhaps this is just so go magic I don't fully understand and there is nothing wrong with this.

**Review by john-a-joyce — APPROVED** [review]
A few comments some more questions than comments so approving.

File: test/e2e/env.go (hunk 2 above)
Comment (mihmatache) [reply test/e2e/env.go:204]: I haven't changed the name of the ENV, I just moved the value to a `const`, as it was used in other places as well. I did that for all ENV values.

File: test/e2e/istio/istio_e2e_suite_test.go (hunk 3 above)
Comment (mihmatache) [reply test/e2e/istio/istio_e2e_suite_test.go:63]: This is just a quick hack so to say, as the only thing currently supported by limited trust in their tests, is the initial check that establishes whether a connection can be established over the virtual mesh. So what I am basically doing is skipping all the other tests except the precondition. As other functionality will be added to limited trust, this should be moved under each test case, and not stay at the suite level.

**Pull request merged**